File: poetry_agent_php/src/Utils/Logger.php
<?php

namespace PoetryAgent\Utils;

use Monolog\Logger as MonologLogger;
use Monolog\Handler\StreamHandler;
use Monolog\Handler\RotatingFileHandler;
use Monolog\Formatter\LineFormatter;
use PoetryAgent\Config\Settings;

/**
 * 日志工具类
 */
class Logger
{
    private static ?MonologLogger $instance = null;

    public static function getInstance(?Settings $settings = null): MonologLogger
    {
        if (self::$instance === null) {
            $settings = $settings ?? new Settings();
            
            $logger = new MonologLogger('poetry_agent');
            
            // 控制台输出
            $consoleHandler = new StreamHandler('php://stdout', self::parseLogLevel($settings->getLogLevel()));
            $consoleHandler->setFormatter(new LineFormatter(
                "[%datetime%] %level_name%: %message% %context% %extra%\n",
                'Y-m-d H:i:s'
            ));
            $logger->pushHandler($consoleHandler);
            
            // 文件输出
            $logFile = $settings->getLogFile();
            if ($logFile) {
                $dir = dirname($logFile);
                if (!is_dir($dir)) {
                    mkdir($dir, 0755, true);
                }
                
                $fileHandler = new RotatingFileHandler($logFile, 7, self::parseLogLevel($settings->getLogLevel()));
                $fileHandler->setFormatter(new LineFormatter(
                    "[%datetime%] %level_name%: %message% %context% %extra%\n",
                    'Y-m-d H:i:s'
                ));
                $logger->pushHandler($fileHandler);
            }
            
            self::$instance = $logger;
        }
        
        return self::$instance;
    }

    private static function parseLogLevel(string $level): int
    {
        $level = strtoupper($level);
        switch ($level) {
            case 'DEBUG':
                return MonologLogger::DEBUG;
            case 'INFO':
                return MonologLogger::INFO;
            case 'NOTICE':
                return MonologLogger::NOTICE;
            case 'WARNING':
                return MonologLogger::WARNING;
            case 'ERROR':
                return MonologLogger::ERROR;
            case 'CRITICAL':
                return MonologLogger::CRITICAL;
            case 'ALERT':
                return MonologLogger::ALERT;
            case 'EMERGENCY':
                return MonologLogger::EMERGENCY;
            default:
                return MonologLogger::INFO;
        }
    }
}

File: poetry_agent/utils/logger.py
"""
日志配置模块
"""
import logging
import sys
from pathlib import Path
from config.settings import settings


def setup_logger(name: str = 'poetry_agent', verbose: bool = False) -> logging.Logger:
    """
    设置日志记录器
    
    Args:
        name: 日志记录器名称
        verbose: 是否启用详细输出
        
    Returns:
        配置好的日志记录器
    """
    logger = logging.getLogger(name)
    
    # 设置日志级别
    if verbose:
        level = logging.DEBUG
    else:
        level = getattr(logging, settings.log_level.upper(), logging.INFO)
    
    logger.setLevel(level)
    
    # 避免重复添加处理器
    if logger.handlers:
        return logger
    
    # 创建格式器
    formatter = logging.Formatter(
        '%(asctime)s - %(name)s - %(levelname)s - %(message)s',
        datefmt='%Y-%m-%d %H:%M:%S'
    )
    
    # 控制台处理器
    console_handler = logging.StreamHandler(sys.stdout)
    console_handler.setLevel(level)
    console_handler.setFormatter(formatter)
    logger.addHandler(console_handler)
    
    # 文件处理器（如果配置了日志文件）
    if settings.log_file:
        log_path = Path(settings.log_file)
        log_path.parent.mkdir(parents=True, exist_ok=True)
        file_handler = logging.FileHandler(log_path, encoding='utf-8')
        file_handler.setLevel(level)
        file_handler.setFormatter(formatter)
        logger.addHandler(file_handler)
    
    return logger

